internal/controller/apiextensions/composite/composed.go

```diff
@@ -159,47 +160,83 @@ func NewAPIConnectionDetailsFetcher(c client.Client) *APIConnectionDetailsFetche
 }
 
 // FetchConnectionDetails of the supplied composed resource, if any.
-func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) {
+func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) { // nolint:gocyclo
 	sref := cd.GetWriteConnectionSecretToReference()
+	var data map[string][]byte
 	if sref == nil {
-		return nil, nil
+		// We hit this block if the composed resource does not specify a secret ref
+		// this would occur if it does not publish any secrets, however we may want to still
+		// publish some values or access values from fieldPaths.
+		data = make(map[string][]byte)
+	} else {
+		// It's possible that the composed resource does want to write a
+		// connection secret but has not yet. We presume this isn't an issue and
+		// that we'll propagate any connection details during a future
+		// iteration.
+		s := &corev1.Secret{}
+		nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
+		if err := cdf.client.Get(ctx, nn, s); client.IgnoreNotFound(err) != nil {
+			return nil, errors.Wrap(err, errGetSecret)
+		}
+		data = s.Data
 	}
-
 	conn := managed.ConnectionDetails{}
 
-	// It's possible that the composed resource does want to write a
-	// connection secret but has not yet. We presume this isn't an issue and
-	// that we'll propagate any connection details during a future
-	// iteration.
-	s := &corev1.Secret{}
-	nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
-	if err := cdf.client.Get(ctx, nn, s); client.IgnoreNotFound(err) != nil {
-		return nil, errors.Wrap(err, errGetSecret)
-	}
-
 	for _, d := range t.ConnectionDetails {
-		if d.Name != nil && d.Value != nil {
-			conn[*d.Name] = []byte(*d.Value)
-			continue
+		switch d.Type {
+		case v1.ConnectionDetailValue:
+			if d.Name != nil && d.Value != nil {
+				// Name must be set if value type
+				conn[*d.Name] = []byte(*d.Value)
+			}
+		case v1.ConnectionDetailFromConnectionSecretKey:
+			if d.FromConnectionSecretKey != nil && data[*d.FromConnectionSecretKey] != nil {
+				// FromConnectionSecretKey must be set if value type
+				key := *d.FromConnectionSecretKey
+				if d.Name != nil {
+					key = *d.Name
+				}
+				if key != "" {
+					conn[key] = data[*d.FromConnectionSecretKey]
+				}
+			}
+		case v1.ConnectionDetailFromResourceFieldPath:
+			if d.Name != nil && d.FromResourceFieldPath != nil {
+				_ = extractFieldPathValue(cd, d, conn)
+			}
 		}
+	}
 
-		if d.FromConnectionSecretKey == nil {
-			continue
-		}
+	if len(conn) == 0 {
+		return nil, nil
+	}
```

Comment:
Does it matter whether we return `nil` vs an empty `managed.ConnectionDetails{}` here?

Reply:
Yup, if we return an empty managed.ConnectionDetails{}, it creates an empty secret.

Reply:
The current behavior is that no secret is created, so I added this check to ensure that aspect of the connectionDetails does not change